stan/math/prim/scal/err/is_greater.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_IS_GREATER_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_IS_GREATER_HPP
+
+#include <stan/math/prim/scal/meta/is_vector_like.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/length.hpp>
+#include <stan/math/prim/scal/meta/get.hpp>
+#include <functional>
+
+namespace stan {
+namespace math {
+
+/**
+ * Check if <code>y</code> is strictly greater than <code>low</code>.
+ * This function is vectorized and will check each element of
+ * <code>y</code> against each element of <code>low</code>.
+ * @tparam T_y Type of <code>y</code>
+ * @tparam T_low Type of lower bound
+ * @param y Variable to check
+ * @param low Lower bound
+ * @return <code>true</code> if <code>y</code> is greater than low and if no
+ *   element of <code>y</code> or <code>low</code> is NaN.
+ */
+template <typename T_y, typename T_low>
```

Comment:
When is a lower bound going to be vectorized like this?  I'm also confused by what look like two different bodies here, one treating `y` like a scalar and one treating it like a vector.  If the `get()` applied to scalars, isn't this just testing the same thing twice since the first loop seems to assume a scalar?